File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.Data.Filtering;
using DevExpress.XtraGrid.Columns;


namespace DXSample {
    public partial class Main: XtraForm {
        public Main() {
            InitializeComponent();
        }

        string fieldToRemove = string.Empty;
        private void OnFormLoad(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.Employees' table. You can move, or remove it, as needed.
            this.employeesTableAdapter.Fill(this.nwindDataSet.Employees);

            PopilateRadioGroup();
        }

        private void PopilateRadioGroup()
        {
            foreach (GridColumn col in gridView1.Columns)
                radioGroup1.Properties.Items.Add(new DevExpress.XtraEditors.Controls.RadioGroupItem(col.FieldName, col.GetCaption()));
        }

        private void OnRemoveFieldChanged(object sender, EventArgs e)
        {
            RadioGroup radioGroup = sender as RadioGroup;
            object val = radioGroup.EditValue;
            fieldToRemove = val.ToString();
        }

        private void OnApplyFilter(object sender, EventArgs e)
        {
            filterControl1.ApplyFilter();
        }

        private void OnRemoveCriteriaByField(object sender, EventArgs e)
        {
            filterControl1.FilterCriteria = CriteriaHelper.RemoveCriteriaByFieldName(fieldToRemove, filterControl1.FilterCriteria);
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Columns

Namespace DXSample

    Public Partial Class Main
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private fieldToRemove As String = String.Empty

        Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs)
            ' TODO: This line of code loads data into the 'nwindDataSet.Employees' table. You can move, or remove it, as needed.
            employeesTableAdapter.Fill(nwindDataSet.Employees)
            PopilateRadioGroup()
        End Sub

        Private Sub PopilateRadioGroup()
            For Each col As GridColumn In gridView1.Columns
                radioGroup1.Properties.Items.Add(New DevExpress.XtraEditors.Controls.RadioGroupItem(col.FieldName, col.GetCaption()))
            Next
        End Sub

        Private Sub OnRemoveFieldChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim radioGroup As RadioGroup = TryCast(sender, RadioGroup)
            Dim val As Object = radioGroup.EditValue
            fieldToRemove = val.ToString()
        End Sub

        Private Sub OnApplyFilter(ByVal sender As Object, ByVal e As EventArgs)
            filterControl1.ApplyFilter()
        End Sub

        Private Sub OnRemoveCriteriaByField(ByVal sender As Object, ByVal e As EventArgs)
            filterControl1.FilterCriteria = RemoveCriteriaByFieldName(fieldToRemove, filterControl1.FilterCriteria)
        End Sub
    End Class
End Namespace
